### PR TITLE
Add Final to multiple methods from the AbstractAdmin

### PR DESCRIPTION
**Pull request by VincentLanglet**
Related to https://github.com/sonata-project/SonataAdminBundle/issues/5791

A lot of public method of the AbstractAdmin shouldn't be extended for different reason
- Some of them have a `protected configureFoo()` method provided for this
- Some of them are calling directly a service `return $this->service->method()`, so the service should be overridden instead.
- It does not make sens for some

Making these method final let me discovering we were often mocking AbstractAdmin instead of the Interface ; so I fixed the tests.

I didn't add final everywhere, this PR is big enough IMHO, I prefer to follow in another one.
Basically I'd like to add final to as much public method of AbstractAdmin as possible, if someone need to configure/override something, it should be done in a configure method or a specific service instead.

### CODE REVIEW

File: src/Admin/AdminHelper.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ class AdminHelper
     /**
      * @var PropertyAccessorInterface
      */
-    protected $propertyAccessor;
+    private $propertyAccessor;
```
Comment (OskarStark) [inline src/Admin/AdminHelper.php:40]: This is a BC break isn't it?

Comment (greg0ire) [reply src/Admin/AdminHelper.php:40]: Yes but the target branch is master, so that should be fine.

Comment (OskarStark) [reply src/Admin/AdminHelper.php:40]: OK I am just thinking that we make it internal in master, which would just raise a warning, but making it private in the same step is another user experience 😄

Comment (OskarStark) [reply src/Admin/AdminHelper.php:40]: Friendly ping @VincentLanglet, thoughts?

Comment (VincentLanglet) [reply src/Admin/AdminHelper.php:40]: The class is declared as internal, people shouldnt use it.

Plus, if we dont do this change on new major, we never do.

Comment (OskarStark) [reply src/Admin/AdminHelper.php:40]: Ok fine

**Review by jordisala1991 — APPROVED** [review]

**Review by wbloszyk — APPROVED** [review]
